internal/controller/apiextensions/composite/composed.go

```diff
@@ -159,47 +163,96 @@ func NewAPIConnectionDetailsFetcher(c client.Client) *APIConnectionDetailsFetche
 }
 
 // FetchConnectionDetails of the supplied composed resource, if any.
-func (cdf *APIConnectionDetailsFetcher) FetchConnectionDetails(ctx context.Context, cd resource.Composed, t v1.ComposedTemplate) (managed.ConnectionDetails, error) {
+func (cdf *APIConnectionDetailsFetcher) FetchConnectionDetails(ctx context.Context, cd resource.Composed, t v1.ComposedTemplate) (managed.ConnectionDetails, error) { // nolint:gocyclo
 	sref := cd.GetWriteConnectionSecretToReference()
+	var data map[string][]byte
 	if sref == nil {
-		return nil, nil
+		// We hit this block if the composed resource does not specify a secret ref
+		// this would occur if it does not publish any secrets, however we may want to still
+		// publish some values or access values from fieldPaths.
+		data = make(map[string][]byte)
+	} else {
+		// It's possible that the composed resource does want to write a
+		// connection secret but has not yet. We presume this isn't an issue and
+		// that we'll propagate any connection details during a future
+		// iteration.
+		s := &corev1.Secret{}
+		nn := types.NamespacedName{Namespace: sref.Namespace, Name: sref.Name}
+		if err := cdf.client.Get(ctx, nn, s); client.IgnoreNotFound(err) != nil {
+			return nil, errors.Wrap(err, errGetSecret)
+		}
+		data = s.Data
 	}
-
 	conn := managed.ConnectionDetails{}
 
-	// It's possible that the composed resource does want to write a
-	// connection secret but has not yet. We presume this isn't an issue and
-	// that we'll propagate any connection details during a future
-	// iteration.
-	s := &corev1.Secret{}
-	nn := types.NamespacedName{Namespace: sref.Namespace, Name: sref.Name}
-	if err := cdf.client.Get(ctx, nn, s); client.IgnoreNotFound(err) != nil {
-		return nil, errors.Wrap(err, errGetSecret)
-	}
-
 	for _, d := range t.ConnectionDetails {
-		if d.Name != nil && d.Value != nil {
-			conn[*d.Name] = []byte(*d.Value)
-			continue
-		}
+		switch d.Type {
+		case v1.ConnectionDetailValue:
+			// Name, Value must be set if value type
+			switch {
+			case d.Name != nil && d.Value != nil:
+				conn[*d.Name] = []byte(*d.Value)
```

Comment:
Nit: I think this could be the `default` case, or just live below the `switch`. Switch cases are evaluated in order, so we'd know the name and value weren't nil if we got past the below two cases.

Reply:
Done!